source/extensions/common/wasm/wasm.cc

```diff
@@ -788,6 +759,74 @@ uint32_t Context::httpCall(absl::string_view cluster, const Pairs& request_heade
   return token;
 }
 
+uint32_t Context::grpcCall(const envoy::api::v2::core::GrpcService& grpc_service,
+                           absl::string_view service_name, absl::string_view method_name,
+                           absl::string_view request,
+                           const absl::optional<std::chrono::milliseconds>& timeout) {
+  auto token = next_grpc_token_++;
+  if (IsGrpcStreamToken(token)) {
+    token = next_grpc_token_++;
+  }
+  // Handle rollover.
+  for (;;) {
+    if (token == 0) {
+      token = next_grpc_token_ += 2;
+    }
+    if (!grpc_call_request_.count(token)) {
+      break;
+    }
+    token = next_grpc_token_ += 2;
+  }
+  auto& handler = grpc_call_request_[token];
+  auto grpc_request =
+      clusterManager()
+          .grpcAsyncClientManager()
+          .factoryForGrpcService(grpc_service, wasm_->scope_, true /* skip_cluster_check */)
+          ->create()
```

Comment:
This seems to be not right. create() returns an unique_ptr of async client and gets deconstructed after function returns, which causes connection gets cleaned up and closed. The unique ptr should be moved into handler.

Reply:
You are correct.  Fixed.